swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/work/WorkQueueManager.kt

```diff
@@ -65,10 +69,8 @@ class WorkQueueManager(
 
   override val onDiscoveryDownCallback: (event: RemoteStatusChangedEvent) -> Unit
     get() = {
-      if (!isEnabled()) {
-        // should only empty the queue if disabled.
-        queue.clear()
-      }
+      up.set(false)
+      queue.clear()
```

Comment:
I don't think this is the right behavior here. With `RedisWorkQueue`, queue state is shared across instances. If we're running 3 swabbie instances and one is disabled, that shouldn't impact the other 2.

Reply:
😌 oh crap you are totally right. I shall fix this.